src/morse_code_decoderii/codewars/m_sulkouski/github/com/MorseCodeDecoder_II.java

```diff
@@ -0,0 +1,179 @@
+//In this kata you have to write a Morse code decoder for wired electrical telegraph.
+//        Electric telegraph is operated on a 2-wire line with a key that, when pressed, connects the wires together, which can be detected on a remote station.
+//        The Morse code encodes every character being transmitted as a sequence of "dots" (short presses on the key) and "dashes" (long presses on the key).
+//
+//        When transmitting the Morse code, the international standard specifies that:
+//
+//        "Dot" – is 1 time unit long.
+//        "Dash" – is 3 time units long.
+//        Pause between dots and dashes in a character – is 1 time unit long.
+//        Pause between characters inside a word – is 3 time units long.
+//        Pause between words – is 7 time units long.
+//        However, the standard does not specify how long that "time unit" is.
+//        And in fact different operators would transmit at different speed. An amateur person may need a few seconds to transmit a single character,
+//        a skilled professional can transmit 60 words per minute, and robotic transmitters may go way faster.
+//
+//        For this kata we assume the message receiving is performed automatically by the hardware that checks the line periodically,
+//        and if the line is connected (the key at the remote station is down), 1 is recorded, and if the line is not connected (remote key is up),
+//        0 is recorded. After the message is fully received, it gets to you for decoding as a string containing only symbols 0 and 1.
+
+package morse_code_decoderii.codewars.m_sulkouski.github.com;
+
+
+public class MorseCodeDecoder_II {
+    public static void main(String[] args) {
+        System.out.println(decodeMorse("1100110011001100000011000000111111001100111111001111110000000000000011001111110011111100111111000000110011001111110000001111110011001100000011"));
+    }
+
+
+    private static String decodeBits(String rawBits) {
+        int currentBitCounter;
+        int index = 0;
+        rawBits = trimBits(rawBits);
```

Comment:
warning, possible danger !!! are you changing the original reference of rawBits ? or is rawBits a variable that only exists inside decodeBits method?